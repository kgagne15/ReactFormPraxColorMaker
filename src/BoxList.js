import { v4 as uuid } from 'uuid';
import React, {useState} from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const INITIAL_STATE = [];

    const [boxes, setBoxes] = useState(INITIAL_STATE);

    const deleteBox = (id) => {
        let boxCopy = [...boxes];
        boxCopy.splice(id, 1)
        setBoxes([...boxCopy])
    }

    const addBox = (newBox) => {
        newBox.width = parseInt(newBox.width)
        newBox.height = parseInt(newBox.height)
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }

    return (
        <>
            <div>
                {boxes.map(({id, color, width, height}) => <Box deleteBox={deleteBox} id={id} key={id} color={color} width={width} height={height} />)}
            </div>
            <NewBoxForm addBox={addBox} />
        </>
    )
}

export default BoxList;