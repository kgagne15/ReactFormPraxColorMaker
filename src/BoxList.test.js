import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function() {
    render(<BoxList />)
});

it("matches snapshot", function() {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
})

//used solution for help with this
it("should add new item", function() {
    const boxList = render(<BoxList/>)
    const width = boxList.getByLabelText("Box Width:")
    const height = boxList.getByLabelText("Box Height:")
    const color = boxList.getByLabelText("Box Color:")

    expect(boxList.queryByText("X")).not.toBeInTheDocument();

    fireEvent.change(color, { target: { value: 'blue' } });
    fireEvent.change(width, { target: { value: 200 } });
    fireEvent.change(height, { target: { value: 200 } });
    const button = boxList.getByText("Add a new box!");
    fireEvent.click(button);

    expect(boxList.queryByText("X")).toBeInTheDocument();
});

it("should remove an item", function() {
    const boxList = render(<BoxList/>)
    const width = boxList.getByLabelText("Box Width:")
    const height = boxList.getByLabelText("Box Height:")
    const color = boxList.getByLabelText("Box Color:")

    expect(boxList.queryByText("X")).not.toBeInTheDocument();

    fireEvent.change(color, { target: { value: 'blue' } });
    fireEvent.change(width, { target: { value: 200 } });
    fireEvent.change(height, { target: { value: 200 } });
    const button = boxList.getByText("Add a new box!");
    fireEvent.click(button);

    expect(boxList.queryByText("X")).toBeInTheDocument();
    const removeBtn = boxList.getByText("X");
    fireEvent.click(removeBtn);
    expect(boxList.queryByText("X")).not.toBeInTheDocument();
})