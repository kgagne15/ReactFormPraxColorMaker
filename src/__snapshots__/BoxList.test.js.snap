// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div />
  <form>
    <label
      for="width"
    >
      Box Width: 
    </label>
    <input
      id="width"
      name="width"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Box Height: 
    </label>
    <input
      id="height"
      name="height"
      type="text"
      value=""
    />
    <label
      for="color"
    >
      Box Color: 
    </label>
    <input
      id="color"
      name="color"
      type="text"
      value=""
    />
    <button>
      Add a new box!
    </button>
  </form>
</DocumentFragment>
`;
